### PR TITLE
Fixing VB OOB validation in draw and drawIndexed for zero stride

### PR DESCRIPTION
**Pull request by jzm-intel**
In this patch vertex buffer OOB validation are fixed to take the zero-stride buffer situation into consideration.

***
<a href="https://pr-preview.s3.amazonaws.com/jzm-intel/gpuweb/pull/2040.html" title="Last updated on Aug 17, 2021, 6:44 AM UTC (915522c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/2040/6831bfa...jzm-intel:915522c.html" title="Last updated on Aug 17, 2021, 6:44 AM UTC (915522c)">Diff</a>

### CODE REVIEW

**Review by Kangz — APPROVED** [review]
Logic LGTM

**Comment by Kangz** [comment]
Uhh what I meant is that we should have a second review at least for the editorial aspect. Please work with @kainino0x and @kvark on that part.

**Comment by jzm-intel** [comment]
> Uhh what I meant is that we should have a second review at least for the editorial aspect. Please work with @kainino0x and @kvark on that part.

Ah sorry, what to do now? Could you please revert this merge or somehow?

**Comment by Kangz** [comment]
I reverted. Please open the PR again!

**Comment by jzm-intel** [comment]
Thanks!

**Comment by jzm-intel** [comment]
It seems that a merged (and reverted) PR can not be re-opened. I am going to open another PR, sorry for the inconvenience!

**Pull request merged**
